### PR TITLE
[15.0][Migration]hr_applicant 

### PR DESCRIPTION
**Pull request by ilakkia-serpentcs**

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
have you made sure there are no conflicts ?
@Chandresh-SerpentCS  @vimalpatelserpentcs pls review

File: hr_applicant/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# See LICENSE file for full copyright and licensing details.
+
+{
+    "name": "Applicant Process",
```
Comment (vimalpatelserpentcs) [inline hr_applicant/__manifest__.py:4]:
```suggestion
    "name": "HR Applicant Process",
```

File: hr_applicant/models/hr_recruitment.py [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+
+    medical_ids = fields.One2many(
+        "hr.applicant.medical.details", "applicant_id", "Medical Ref."
+    )
+    no_of_medical = fields.Integer(
+        "No of Medical Detials", compute="_compute_no_of_medical", readonly=True
+    )
+    no_of_medical1 = fields.Integer(
+        "No of Medical Detials", compute="_compute_no_of_medical", readonly=True
+    )
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:53]: Why?

File: hr_applicant/models/hr_recruitment.py [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+                            "medical_examination": medical_detail.medical_examination,
+                            "vital_sign": medical_detail.vital_sign,
+                            "date": medical_detail.date,
+                            "doc_comment": medical_detail.doc_comment,
+                            "head_face_scalp": medical_detail.head_face_scalp,
+                            "nose_sinuses": medical_detail.nose_sinuses,
+                            "mouth_throat": medical_detail.mouth_throat,
+                            "ears_tms": medical_detail.ears_tms,
+                            "eyes_pupils_ocular": medical_detail.eyes_pupils_ocular,
+                            "heart_vascular_system": medical_detail.heart_vascular_system,
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:147]: We need to improve this method code...like use prepare method to separate each object records vals

def _prepare_employee_vals():
    //code

vals = self._prepare_employee_vals()

File: hr_applicant/models/hr_recruitment.py [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+                                "res_id": language_id.id}
+                        )
+        return res
+
+
+class ApplicantMedicalDetails(models.Model):
+
+    _name = "hr.applicant.medical.details"
+    _description = "Applicant Medical Details"
+    _rec_name = "medical_examination"
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:342]: Would be better to make a separate .py file each object

File: hr_applicant/models/hr_recruitment.py [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+    occupation = fields.Char(string="Occupation", size=128)
+    gender = fields.Selection(
+        [("Male", "Male"), ("Female", "Female")], string="Gender", required=False
+    )
+    active = fields.Boolean(string="Active", default=True)
+    applicant_id = fields.Many2one(
+        "hr.applicant", "Applicant Ref", ondelete="cascade")
+
+    @api.onchange("birthday")
+    def onchange_birthday(self):
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:498]: should be private method

File: hr_applicant/models/hr_recruitment.py [hunk 6]
```diff
@@ -0,0 +1,710 @@
...
+            rec.from_date = False
+            rec.to_date = False
+            rec.education_rank = ""
+            rec.school_name = ""
+            rec.grade = ""
+            rec.field = ""
+            rec.edu_type = ""
+            rec.country_id = False
+            rec.state_id = False
+            rec.province = ""
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:583]: use .update method

File: hr_applicant/models/hr_recruitment.py [hunk 7]
```diff
@@ -0,0 +1,710 @@
...
+    @api.onchange("from_date", "to_date")
+    def onchange_date(self):
+        to_date = self.to_date and datetime.strftime(
+            self.to_date, DEFAULT_SERVER_DATE_FORMAT
+        )
+        if (
+            to_date
+            and datetime.strptime(to_date, DEFAULT_SERVER_DATE_FORMAT)
+            >= datetime.today()
+        ):
```
Comment (vimalpatelserpentcs) [inline hr_applicant/models/hr_recruitment.py:602]: For Datetime conversion, Please use fields. Datetime class method.
like from_string, to_string method

**Review by vimalpatelserpentcs — COMMENTED** [review]
Improvement needed:
1) all onchange methods should be a private method
2) for Datetime conversion: use please fields.Datetime class method's (from_string,to_string)
3) make separate .py file for each object

**Pull request closed without merging**